libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -179,6 +181,11 @@ private static PolicyManager createPolicyManager() {
         );
     }
 
+    private static @Nullable Path getUserHome() {
+        String userHome = System.getProperty("user.home");
```

Comment:
@rjernst regarding your question, I'd assume this is always set. However, the following code in `javax.activation.MimetypesFileTypeMap` (which requires home relative file permissions) made me go for a null-safe impl

https://github.com/javaee/activation/blob/270afe5ae592c65b1c944c6a882c1720574fbf19/activation/src/main/java/javax/activation/MimetypesFileTypeMap.java#L132-L139

Reply:
What does it mean in the ES case though? We can't provide a null path, it doesn't make sense. Can we throw an assertion if it is null? The link is some weird part of javax.activiation, it's not in the base jdk.

Reply:
any more comments @rjernst ?

Reply:
I'll proceed merging @rjernst, let me know if you have more comments and I'll follow up.